File: backend/tests/fixtures/parsers/pascal/hello.pas
program HelloDemo;

type
    TGreeter = class
        procedure Greet(name: string);
        procedure FormatName(n: string);
    end;

procedure TGreeter.Greet(name: string);
begin
    FormatName(name);
end;

procedure TGreeter.FormatName(n: string);
begin
    WriteLn(n);
end;

begin
    WriteLn('start');
end.

File: backend/tests/fixtures/parsers/fortran/hello.f90
module greeter_mod
    implicit none
contains
    subroutine greet(name)
        character(len=*), intent(in) :: name
        call format_name(name)
    end subroutine greet

    subroutine format_name(n)
        character(len=*), intent(in) :: n
        print *, n
    end subroutine format_name
end module greeter_mod
